src/sorcha/modules/PPOutput.py

```diff
@@ -92,20 +95,37 @@ def PPOutWriteSqlite3(pp_results, outf, lastchunk=False, tablename="sorcha_resul
 
     pp_results.to_sql(tablename, con=cnx, if_exists="append", index=False)
 
-    # we don't want to index the table until we're sure we're done appending to it
-    # as recreating the indexes on every append is slow
-    if lastchunk:
-        pplogger.info("Last chunk detected. Indexing SQL table...")
-        cur = cnx.cursor()
-        cur.execute("CREATE INDEX ObjID ON {} (ObjID)".format(tablename))
-        cur.execute("CREATE INDEX fieldMJD_TAI ON {} (fieldMJD_TAI)".format(tablename))
-        cur.execute("CREATE INDEX optFilter ON {} (optFilter)".format(tablename))
-        cnx.commit()
-
     pplogger.info("SQL results saved in table {} in database {}.".format(tablename, outf))
 
 
-def PPWriteOutput(cmd_args, configs, observations_in, endChunk=0, verbose=False, lastchunk=False):
+def PPIndexSQLDatabase(outf, tablename="sorcha_results"):
+    """
+    Indexes a SQLite database of Sorcha output.
+
+    Parameters
+    -----------
+    outf : string
+        Location of SQLite database to be indexed.
+
+    tablename: string
+        String of the table within the database to be indexed.
+
+    Returns
+    -----------
+    None.
+
+    """
+
+    cnx = sqlite3.connect(outf)
```

Comment:
What if all the chunks are empty and it never ever got here. I'd add a quick check that the file actually exists or just gracefully return out.